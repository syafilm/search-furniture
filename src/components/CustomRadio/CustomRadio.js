import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

import {
  Styles,
} from 'utils'

const CustomRadio = ({
  label,
  value,
  onChange,
  text,
  color,
  checked,
}) => (
  <Fragment>
    <input
      value={value}
      onChange={onChange}
      type="radio"
      checked={checked}
      className={Styles.formControlRadio(color)}
      id={text}
    />
    <label htmlFor={text} className={Styles.labelRadio}>{label}</label>
  </Fragment>
)

CustomRadio.propTypes = {
  label: PropTypes.string,
  checked: PropTypes.bool,
  value: PropTypes.string,
  onChange: PropTypes.func,
  text: PropTypes.string,
  color: PropTypes.string,
}

CustomRadio.defaultProps = {
  label: '',
  value: '',
  text: '',
  onChange: null,
  color: '',
  checked: false,
}

export default CustomRadio
