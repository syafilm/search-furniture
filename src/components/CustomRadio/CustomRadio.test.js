import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import CustomRadio from './CustomRadio'

describe('Component CustomRadio', () => {
  const mockEvent = jest.fn()
  const trending = 'trending'
  const colorReverse = '#000'
  const label = 'Trending'
  it('Can render CustomRadio with checked', () => {
    const component = shallow(
      <CustomRadio
        text={trending}
        onChange={mockEvent}
        checked={trending === 'trending'}
        value={trending}
        color={colorReverse}
        label={label}
      />,
    )
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
    expect(component.find('div')).toBeDefined()
  })

  it('Can render CustomRadio with unchecked', () => {
    const component = shallow(
      <CustomRadio
        text={trending}
        onChange={mockEvent}
        checked={trending === ''}
        value={trending}
        color={colorReverse}
        label={label}
      />,
    )
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
    expect(component.find('div')).toBeDefined()
  })
})
