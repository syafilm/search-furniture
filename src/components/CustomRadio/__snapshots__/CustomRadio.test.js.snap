// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component CustomRadio Can render CustomRadio with checked 1`] = `
<Fragment>
  <input
    checked={true}
    className="css-1bfy9hx"
    id="trending"
    onChange={[MockFunction]}
    type="radio"
    value="trending"
  />
  <label
    htmlFor="trending"
  >
    Trending
  </label>
</Fragment>
`;

exports[`Component CustomRadio Can render CustomRadio with unchecked 1`] = `
<Fragment>
  <input
    checked={false}
    className="css-1bfy9hx"
    id="trending"
    onChange={[MockFunction]}
    type="radio"
    value="trending"
  />
  <label
    htmlFor="trending"
  >
    Trending
  </label>
</Fragment>
`;
